libraries/AP_Logger/AP_Logger_SITL.cpp

```diff
@@ -26,14 +26,14 @@ extern const AP_HAL::HAL& hal;
 void AP_Logger_SITL::Init()
 {
     if (flash_fd == 0) {
-        flash_fd = open("dataflash.bin", O_RDWR|O_CLOEXEC, 0777);
+        flash_fd = open(AP_Logger_SITL::filename, O_RDWR|O_CLOEXEC, 0777);
         if (flash_fd == -1) {
-            flash_fd = open("dataflash.bin", O_RDWR | O_CREAT | O_CLOEXEC, 0777);
+            flash_fd = open(AP_Logger_SITL::filename, O_RDWR | O_CREAT | O_CLOEXEC, 0777);
```

Comment:
Don't need to namespace filename here.

Reply:
can't check for nulls with a reference, they need to stay for now. We can update it on another PR.

Reply:
fixed